src/coins.h

```diff
@@ -174,8 +167,10 @@ struct CCoinsCacheEntry
         SetClean();
     }
 
-    static void SetDirty(CoinsCachePair& pair, CoinsCachePair& sentinel) noexcept { AddFlags(DIRTY, pair, sentinel); }
-    static void SetFresh(CoinsCachePair& pair, CoinsCachePair& sentinel) noexcept { AddFlags(FRESH, pair, sentinel); }
+    static void SetDirty(CoinsCachePair& pair, CoinsCachePair& sentinel, bool fresh = false) noexcept
+    {
+        AddFlags(fresh ? FRESH | DIRTY : DIRTY, pair, sentinel);
+    }
```

Comment:
Since it's only called here, we could inline and simplify it now:
```suggestion
    static void SetDirty(CoinsCachePair& pair, CoinsCachePair& sentinel, bool fresh = false) noexcept
    {
        if (!pair.second.m_flags) {
            Assume(!pair.second.m_prev && !pair.second.m_next);
            pair.second.m_prev = sentinel.second.m_prev;
            pair.second.m_next = &sentinel;
            sentinel.second.m_prev = &pair;
            pair.second.m_prev->second.m_next = &pair;
        }
        Assume(pair.second.m_prev && pair.second.m_next);
        pair.second.m_flags |= fresh ? FRESH | DIRTY : DIRTY;
    }
```

----

Not sure, but do the flags still make sense after this or would a `bool is_fresh` also suffice? The dirtiness can be deduced from the `m_next`/`m_prev` pair so the only unknown is the freshness which doesn't necessitate a flag anymore - right?